cirq/pasqal/pasqal_qubits.py

```diff
@@ -192,3 +193,194 @@ def __rsub__(self, other: Tuple[int, int, int]) -> 'ThreeDGridQubit':
 
     def __neg__(self) -> 'ThreeDGridQubit':
         return ThreeDGridQubit(row=-self.row, col=-self.col, lay=-self.lay)
+
+
+class ThreeDQubit(cirq.ops.Qid):
+    """A qubit in 3d.
+
+    ThreeDQubits use z-y-x ordering:
+
+        ThreeDQubit(0, 0, 0) < ThreeDQubit(1, 0, 0)
+        < ThreeDQubit(0, 1, 0) < ThreeDQubit(1, 1, 0)
+        < ThreeDQubit(0, 0, 1) < ThreeDQubit(1, 0, 1)
+        < ThreeDQubit(0, 1, 1) < ThreeDQubit(1, 1, 1)
+
+    New ThreeDQubit can be constructed by adding or subtracting tuples
+
+        >>> cirq.pasqal.ThreeDQubit(2.5, 3, 4.7) + (3, 1.2, 6)
+        pasqal.ThreeDQubit(5.5, 4.2, 10.7)
+
+        >>> cirq.pasqal.ThreeDQubit(2.4, 3.1, 4) - (1, 2, 2.1)
+        pasqal.ThreeDQubit(1.4, 1.1, 1.9)
+    """
+
+    def __init__(self, x: float, y: float, z: float):
+        self.x = x
+        self.y = y
+        self.z = z
+
+    def _comparison_key(self):
+        return round(self.z, 9), round(self.y, 9), round(self.x, 9)
```

Comment:
Why rounding the coordinates? And why to 9 digits?

Reply:
I don't recall exactly what prompted this, but I'm assuming it was some machine precision error. We end up using a lot of irrational numbers from triangular lattice coordinates, so I'm assuming there was some false negative.
As for the rounding, I think it is arbitrary, it could have more digits. I'm happy to change it to a higher value.

Reply:
Great - if you could recover that use case that would be useful - we should add it to the tests to protect this functionality. Otherwise someone might just remove this without failing tests.

Reply:
Turns out rounding to 9 was excessive, I increased the number of digits to 15 and introduced a test to back that up.